tests/Language/Haskell/Stylish/Config/Tests.hs

```diff
@@ -0,0 +1,144 @@
+module Language.Haskell.Stylish.Config.Tests
+    ( tests
+    ) where
+
+
+--------------------------------------------------------------------------------
+import           Control.Exception               hiding (assert)
+import qualified Data.Set                        as Set
+import           System.Directory
+import           System.FilePath                 ((</>))
+import           System.IO.Error
+import           System.Random
+import           Test.Framework                  (Test, testGroup)
+import           Test.Framework.Providers.HUnit  (testCase)
+import           Test.HUnit                      (Assertion, assert)
+--------------------------------------------------------------------------------
+import           Language.Haskell.Stylish.Config
+
+--------------------------------------------------------------------------------
+tests :: Test
+tests = testGroup "Language.Haskell.Stylish.Config"
+    [ testCase "Extensions extracted correctly from .cabal file"
+               testExtensionsFromDotCabal
+    , testCase "Extensions extracted correctly from .stylish-haskell.yaml file"
+               testExtensionsFromDotStylish
+    , testCase "Extensions extracted correctly from .stylish-haskell.yaml and .cabal files"
+               testExtensionsFromBoth
+    ]
+--------------------------------------------------------------------------------
+
+-- | Create a temporary directory with a randomised name built from the template provided
+createTempDirectory :: String -> IO FilePath
+createTempDirectory template  = do
+  tmpRootDir <- getTemporaryDirectory
+  dirId <- randomIO :: IO Word
+  findTempName tmpRootDir dirId
+  where
+    findTempName :: FilePath -> Word -> IO FilePath
+    findTempName tmpRootDir x = do
+      let dirpath = tmpRootDir </> template ++ show x
+      r <- try $ createDirectory dirpath
+      case r of
+        Right _ -> return dirpath
+        Left  e | isAlreadyExistsError e -> findTempName tmpRootDir (x+1)
+                | otherwise              -> ioError e
+
+-- | Perform an action inside a temporary directory tree and purge the tree afterwords
+withTestDirTree :: IO a -> IO a
+withTestDirTree action = do
+  current <- getCurrentDirectory
+  temp <- createTempDirectory "stylish_haskell"
+  setCurrentDirectory temp
+  result <- action
+  setCurrentDirectory current
+  removeDirectoryRecursive temp
+  pure result
```

Comment:
Nitpick: you can use `bracket` to clean this up even in case there's exceptions:

```haskell
withTestDirTree :: IO a -> IO a
withTestDirTree action = bracket
    ((,) <$> getCurrentDirectory <*> createTempDirectory "stylish_haskell")
    (\(current, temp) -> do
        setCurrentDirectory current
        removeDirectoryRecursive temp)
    (\(_, temp) -> setCurrentDirectory temp >> action)
```

Reply:
Indeed! I have refactored `withTestDirTree` to use `bracket` and Applicative notation.